src/Compilers/Core/Portable/DiagnosticAnalyzer/Suppression.cs

```diff
@@ -48,5 +50,34 @@ public static Suppression Create(SuppressionDescriptor descriptor, Diagnostic su
         /// Diagnostic suppressed by this suppression.
         /// </summary>
         public Diagnostic SuppressedDiagnostic { get; }
+
+        public static bool operator ==(Suppression left, Suppression right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(Suppression left, Suppression right)
+        {
+            return !(left == right);
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is Suppression suppression
+                && Equals(suppression);
+        }
+
+        public bool Equals(Suppression other)
+        {
+            return EqualityComparer<SuppressionDescriptor>.Default.Equals(Descriptor, other.Descriptor)
+                && EqualityComparer<Diagnostic>.Default.Equals(SuppressedDiagnostic, other.SuppressedDiagnostic);
+        }
+
+        public override int GetHashCode()
+        {
+            return Hash.Combine(
+                EqualityComparer<SuppressionDescriptor>.Default.GetHashCode(Descriptor),
```

Comment:
>GetHashCode(Descriptor) [](start = 64, length = 23)

According to documentation, this is going to throw on a null input. Should we protect?

Reply:
No. The documentation is wrong. This is the long-time safe use of this type. See dotnet/runtime#30998.